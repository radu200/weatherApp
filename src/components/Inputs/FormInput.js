import React from "react";
import PropTypes from 'prop-types';


const FormInput = ({
  name,
  type,
  placeholder,
  handleChange,
  value,
  className
}) => {
  return (
    <input
      data-test-id="input-search"
      className={className}
      name={name}
      type={type}
      placeholder={placeholder}
      onChange={handleChange}
      value={value}
    />
  );
};

FormInput.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  value: PropTypes.any,
  className:PropTypes.string,
  onChange: PropTypes.func
}
export default FormInput;