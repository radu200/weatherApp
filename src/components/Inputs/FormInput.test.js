import React from 'react'
import FormInput from './FormInput'
import { shallow, mount } from 'enzyme'     
import { shallowToJson } from 'enzyme-to-json'



describe('FormInput component', () => {
  const props = {
    name:'some name',
    type:'text',
    placeholder:'search',
    className:'some class'
  }
  let wrap;
   beforeEach(() => {
      wrap = shallow(<FormInput {...props}/>)
   })

  it('should render FormInput with given props', () => {
    expect(shallowToJson(wrap)).toMatchSnapshot()
})
    
    it('should call onChange event', () => {
       const onSearch = jest.fn() 
       const component = shallow(<FormInput handleChange={onSearch} />)
       const input = component.find('input');
       input.simulate('change')
       expect(onSearch).toHaveBeenCalled()
    })
})
