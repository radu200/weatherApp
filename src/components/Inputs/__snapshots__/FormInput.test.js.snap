// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render FormInput with given props 1`] = `
<input
  className="some class"
  data-test-id="input-search"
  name="some name"
  placeholder="search"
  type="text"
/>
`;
